src/main/java/se/liu/ida/hefquin/engine/queryplan/physical/impl/PhysicalOpBindJoinWithUNION.java

```diff
@@ -0,0 +1,38 @@
+package se.liu.ida.hefquin.engine.queryplan.physical.impl;
+
+import se.liu.ida.hefquin.engine.federation.FederationMember;
+import se.liu.ida.hefquin.engine.federation.SPARQLEndpoint;
+import se.liu.ida.hefquin.engine.queryplan.ExpectedVariables;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.ExecOpBindJoinSPARQLwithUNION;
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.ops.UnaryExecutableOp;
+import se.liu.ida.hefquin.engine.queryplan.logical.UnaryLogicalOp;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpTPAdd;
+
+public class PhysicalOpBindJoinWithUNION extends BasePhysicalOpSingleInputJoin {
+
+	public PhysicalOpBindJoinWithUNION( final UnaryLogicalOp lop) {
+		super(lop);
+	}
+
+	@Override
+	public UnaryExecutableOp createExecOp(final ExpectedVariables... inputVars) {
+		for (final ExpectedVariables ev : inputVars) {
+			System.out.println(ev.getCertainVariables());
+			if (ev.getPossibleVariables().size() > 0) {
+				throw new IllegalArgumentException("Illegal child operator for tpAdd. Possible variables should be empty.");
```

Comment:
```suggestion
				// The executable operator for this physical operator (i.e., ExecOpBindJoinSPARQLwithUNION)
				// can work correctly only in cases in which all input solution mappings are for the exact
				// same set of variables. This can be guaranteed only if the set of possible variables from
				// the child operator is empty.
				throw new IllegalArgumentException("Nonempty set of possible variables.");
```

Reply:
In contrast to your exception message, it is *not* generally necessary that the child of a tpAdd has an empty set of possible variables. In contrast, that's only necessary if we want to implement the (logical) tpAdd operator by using the ExecOpBindJoinSPARQLwithUNION.